proc_macros/src/rule.rs

```diff
@@ -393,6 +400,16 @@ fn get_rule_rule_impl(
                 Some(capture_name) => quote!(Some(#capture_name.into())),
                 None => quote!(None),
             });
+    let maybe_deserialize_options = match rule.options_type.as_ref() {
+        None => quote!(),
+        Some(options_type) => {
+            quote! {
+                let options: #options_type = options.map(|options| {
+                    #crate_name::serde_json::from_str(&options.to_string()).expect("Couldn't parse rule options")
+                }).unwrap();
```

Comment:
I'd think that a rule (using the `rule!` macro, see below what the syntax for this is currently) should be able to declare its options as "optional" by saying `options_type: Option<whatever>`?

And so then here I think it would just conditionally not do this `.unwrap()` (or really the nicer-error-message version of that) if it "sniffs" (here at the proc-macro stage) that the provided `options_type` is in fact an `Option<whatever>`?